src/coreclr/jit/morph.cpp

```diff
@@ -14157,6 +14180,19 @@ void Compiler::fgMergeBlockReturn(BasicBlock* block)
             }
         }
 
+#ifdef SWIFT_SUPPORT
+        // If merging GT_SWIFT_ERROR_RET nodes, ensure the error operand is stored to the merged return error local,
+        // so the correct error value is retrieved in the merged return block.
+        if (errorVal != nullptr)
+        {
+            assert(genReturnErrorLocal != BAD_VAR_NUM);
+            Statement* insertAfterStmt = block->lastStmt();
+            GenTree* swiftErrorStore = gtNewTempStore(genReturnErrorLocal, errorVal, CHECK_SPILL_NONE, &insertAfterStmt,
+                                                      insertAfterStmt->GetDebugInfo(), block);
+            fgNewStmtAtEnd(block, swiftErrorStore);
+        }
```

Comment:
Doesn't this end up reordering `op1` and `op2` of the `GT_SWIFT_ERROR_RET`?

Reply:
In terms of the order of stores, I think so -- is there a correctness issue to doing that? I guess we can insert this new statement before the last one to fix the ordering.

Reply:
The stores themselves aren't the problem, but rather it's the fact that `errorVal` can have side effects. For example, you can imagine something like
```csharp
*err = new SwiftError((void*)Foo(1));
return Foo(2);
```
If we have `GT_SWIFT_ERROR_RET(Foo(1), Foo(2)))` then we would end up with the `Foo(2)` call before the `Foo(1)` call.